src/service/url-expander/expander.js

```diff
@@ -106,7 +112,14 @@ export class Expander {
           } else {
             // or the global source
             binding = this.variableSource_.get(match.name);
-            binding = binding.async || binding.sync;
+            if (opt_sync && binding.sync) {
```

Comment:
can we move this logic into `evaluateBinding`? so that we have all sync/async handling in one place.

Reply:
talked offline. This is still tricky because it needs access to `opt_bindings` and also `matches`, how about a separate method that maps the binding?